mshadow/base.h

```diff
@@ -678,6 +678,41 @@ struct minimum {
     LOG(FATAL) << "Unknown type enum " << type;       \
   }
 
+#define MSHADOW_REAL_TYPE_SWITCH_EX(type, DType, DLargeType, ...)  \
+  switch (type) {                                   \
+  case mshadow::kFloat32:                           \
+    {                                               \
+      typedef float DType;                          \
+      typedef double DLargeType;                    \
+      {__VA_ARGS__}                                 \
+    }                                               \
+    break;                                          \
+  case mshadow::kFloat64:                           \
+    {                                               \
+      typedef double DType;                         \
+      typedef double DLargeType;                    \
+      {__VA_ARGS__}                                 \
+    }                                               \
+    break;                                          \
+  case mshadow::kFloat16:                           \
+    {                                               \
+      typedef mshadow::half::half_t DType;          \
+      typedef float DLargeType;                     \
```

Comment:
Not sure what you mean -- this isn't use anywhere else so far.  What backward compatibility might be broken?